File: db4o.cs/src/com/db4o/cs/foundation/NetworkSocket.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */

package com.db4o.cs.foundation;

import java.io.*;
import java.net.*;

public class NetworkSocket extends NetworkSocketBase {

    public NetworkSocket(String hostName, int port) throws IOException {
    	super(new Socket(hostName, port), hostName);
    }

	public NetworkSocket(Socket socket) throws IOException {
		super(socket);
	}

	@Override
	protected Socket4 createParallelSocket(String hostName, int port) throws IOException {
		return new NetworkSocket(hostName, port);
	}

}

File: db4o.net/Db4objects.Db4o.CS/Db4objects.Db4o.CS/CS/Foundation/NetworkSocket.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.CS.Foundation;

namespace Db4objects.Db4o.CS.Foundation
{
	public class NetworkSocket : NetworkSocketBase
	{
		/// <exception cref="System.IO.IOException"></exception>
		public NetworkSocket(string hostName, int port) : base(new Sharpen.Net.Socket(hostName
			, port), hostName)
		{
		}

		/// <exception cref="System.IO.IOException"></exception>
		public NetworkSocket(Sharpen.Net.Socket socket) : base(socket)
		{
		}

		/// <exception cref="System.IO.IOException"></exception>
		protected override ISocket4 CreateParallelSocket(string hostName, int port)
		{
			return new Db4objects.Db4o.CS.Foundation.NetworkSocket(hostName, port);
		}
	}
}
